nim_status/client.nim

```diff
@@ -90,58 +153,228 @@ proc storeDerivedAccountsInDb(self: StatusObject, id: UUID, keyUid: string,
+# TODO: Remove this from the client if not needed. This is only used for tests
+# right now.
+proc createSettings*(self: StatusObject, settings: Settings,
+  nodeConfig: JsonNode): CreateSettingsResult =
 
-  self.accountsGenerator.loadAccount(address, password, dir)
+  if not self.isLoggedIn:
+    return CreateSettingsResult.err "Not logged in. You must be logged in to " &
+      "create settings."
+  try:
+    self.userDb.createSettings(settings, nodeConfig)
+    return CreateSettingsResult.ok
+  except Exception as e:
+    return CreateSettingsResult.err e.msg
 
-proc closeUserDB*(self: StatusObject) =
-  self.userDb.close()
+proc getPublicAccounts*(self: StatusObject): seq[PublicAccount] =
+  self.accountsDb.getPublicAccounts()
+
+proc toWalletAccount(account: accounts.Account): WalletAccount =
+  let name = if account.name.isNone: "" else: account.name.get
+  WalletAccount(address: account.address, name: name)
+
+proc getWalletAccounts*(self: StatusObject): WalletAccountsResult =
+  if not self.isLoggedIn:
+    return WalletAccountsResult.err "Not logged in. Must be logged in to get " &
+      "wallet accounts."
+  try:
+    let accounts = self.userDb.getWalletAccounts().map(a => a.toWalletAccount)
+    return WalletAccountsResult.ok accounts
+  except Exception as e:
+    return WalletAccountsResult.err "Error getting wallet accounts: " & e.msg
+
+# proc getSetting*[T](self: StatusObject, U: typedesc[T],
+#   setting: SettingsCol): GetSettingResult[T] =
+
+#   if not self.isLoggedIn:
+#     return GetSettingResult.err "Not logged in. Must be logged in to get " &
+#       "settings."
+#   try:
+#     let opt = self.userDb.getSetting(U, setting)
+#     if opt.isNone:
+#       return GetSettingResult.err "asdf"
+#     return GetSettingResult.ok opt.get
+#   except Exception as e:
+#     return GetSettingResult.err e.msg
+
+# proc getSetting*[T](self: StatusObject, U: typedesc[T], setting: SettingsCol,
+#   defaultValue: T): GetSettingResult[T] =
+
+#   if not self.isLoggedIn:
+#     return GetSettingResult.err "Not logged in. Must be logged in to get " &
+#       "settings."
+#   try:
+#     let opt = self.userDb.getSetting(U, setting)
+#     if opt.isNone:
+#       return GetSettingResult.err "asdf"
+#     return GetSettingResult.ok opt.get
+#   except Exception as e:
+#     return GetSettingResult.err e.msg
+
+proc getSettings*(self: StatusObject): GetSettingsResult =
+  if not self.isLoggedIn:
+    return GetSettingsResult.err "Not logged in. Must be logged in to get " &
+      "settings."
+  try:
+    return GetSettingsResult.ok self.userDb.getSettings()
+  except Exception as e:
+    return GetSettingsResult.err e.msg
+
+proc importMnemonic*(self: StatusObject, mnemonic: Mnemonic,
+  bip39Passphrase, password: string, dir: string): PublicAccountResult =
+
+  try:
+    let
+      imported = ?self.accountsGenerator.importMnemonic(mnemonic,
+        bip39Passphrase)
+      paths = @[PATH_WALLET_ROOT, PATH_EIP_1581, PATH_WHISPER,
```

Comment:
These paths are also declared in `createAccount`, maybe makes sense to extract it into a `const` or something.

Reply:
Hmm all the paths are already constants, but i do see how these four paths are used together in two spots. I feel like pulling more out in to constants isn't really going to benefit us too much here. I'd prefer to leave it as is.